### PR TITLE
feat: added did register to dashboard

### PR DESCRIPTION
**Pull request by Dudleyneedham**
## fixes https://github.com/KILTprotocol/ticket/issues/1038

Added button to register and delete DID on the dashboard.
Change the store management of selected Identities with the wallet.
Fixed issues of the selected wallet being undefined. Now throws an error if selected identity not handled.

## How to test:

Add and delete a did within the dashboard.

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: src/containers/WalletAdd/WalletAdd.tsx [hunk 1]
```diff
@@ -96,6 +96,7 @@ class WalletAdd extends React.Component<Props, State> {
       Contacts.Store.addContact(
         ContactRepository.getContactFromIdentity(newIdentity, {
           unregistered: true,
+          addedAt: Date.now(),
```
Comment (tjwelde) [inline src/containers/WalletAdd/WalletAdd.tsx:99]: Why was this added?

**Review by tjwelde — CHANGES_REQUESTED** [review]
Looking pretty good already

File: src/containers/WalletAdd/WalletAdd.tsx (hunk 1 above)
Comment (Dudleyneedham) [reply src/containers/WalletAdd/WalletAdd.tsx:99]: Its the fix in the develop branch, I will remove the commit.

It shouldn't be inside.
=

File: src/services/ClientVersionHelper.ts [hunk 2]
```diff
@@ -20,9 +20,13 @@ class ClientVersionHelper {
...
       resetCause.firstBlockHashChanged = true
     } else {
-      const selectedIdentity: Wallet.Entry = Wallet.getSelectedIdentity(
+      const selectedIdentity = Wallet.getSelectedIdentity(
         persistentStoreInstance.store.getState()
-      )
+      )?.identity
+
+      if (!selectedIdentity) {
+        throw new Error('No selected Identity')
```
[the remark this answers is not in the document]
Comment (Dudleyneedham) [reply src/services/ClientVersionHelper.ts:28]: To be honest, I was thinking of completely removing the `else` block. It doesn't do anything.

[the remark this answers is not in the document]
Comment (tjwelde) [reply src/services/ClientVersionHelper.ts:28]: Just remove your `if (!selectedIdentity)` conditional, since it is unnecessary here.

**Review by tjwelde — APPROVED** [review]
LGTM
